src/components/Dashboard/AddBookmarkModal.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+
+import { Formik } from 'formik';
+import TextInputGroup from '../TextInputGroup';
+import useDerivedStateFromProps from '../../hooks/useDerivedStateFromProps';
+
+const AddBookmarkModal = ({ setAddBookmark, currentBookmark, handleAddBookmark }) => {
+	const [bookmark, setBookmark] = useDerivedStateFromProps(currentBookmark);
+	const [error, setError] = useState();
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+
+		let error;
+
+		if (bookmark.url !== currentBookmark.url) {
+			error = handleAddBookmark({ ...bookmark, createdAt: new Date().toLocaleString() });
+		} else {
+			error = handleAddBookmark(currentBookmark);
+		}
+
+		if (error.type === 'success') {
```

Comment:
error.type === 'success' is een beetje vreemd ;)
Ik zou die `error` hernoemen naar `result`